src/main/java/org/vanilladb/core/storage/metadata/index/IndexMgr.java

```diff
@@ -145,4 +146,65 @@ public void createIndex(String idxName, String tblName, String fldName,
 		iiMap.put(tblName, result);
 		return result;
 	}
+
+	/**
+	 * Returns the requested index.
+	 * 
+	 * @param idxName
+	 *            the name of the index
+	 * @param tx
+	 *            the calling transaction
+	 * @return an Index object
+	 */
+	public Index getIndex(String idxName, Transaction tx) {
+		String tblname = null, fldname = null;
+		RecordFile rf = ti.open(tx, true);
+		rf.beforeFirst();
+		while (rf.next()) {
+			if (rf.getVal(ICAT_IDXNAME).equals(new VarcharConstant(idxName))) {
+				tblname = (String) rf.getVal(ICAT_TBLNAME).asJavaVal();
+				fldname = (String) rf.getVal(ICAT_FLDNAME).asJavaVal();
+				break;
+			}
+		}
+		rf.close();
+
+		IndexInfo ii;
+		Map<String, IndexInfo> iiTable;
+		if (tblname == null || fldname == null)
+			return null;
+		if ((iiTable = iiMap.get(tblname)) == null || (ii = iiTable.get(fldname)) == null)
```

Comment:
There are several problems here:

- Since `iiMap` is a cache, there might be a case that the index catalog file has this `IndexInfo`, but `iiMap` doesn't. Therefore, this implementation may miss those `IndexInfo`s in the case.
- Since you have iterated the catalog file in a few lines of code above, if you found the matched index in the catalog file, there would be no need for it to check `iiMap`.

Here is the flow we plan to implement:

1. Iterate the catalog file to find out the required `IndexInfo`
2. If it didn't found it, returns `null`.
3. Update `iiMap` for fast retrieval in the future

You can leave this `TODO` for us by replacing the code in this method with a comment `TODO: Implement it`.